store/principal.go

```diff
@@ -96,15 +200,15 @@ func (s *PrincipalService) FindPrincipal(ctx context.Context, find *api.Principa
 	}
 	defer tx.PTx.Rollback()
 
-	list, err := findPrincipalList(ctx, tx.PTx, find)
+	list, err := findPrincipalRawListImpl(ctx, tx.PTx, find)
 	if err != nil {
 		return nil, err
 	}
 
 	if len(list) == 0 {
-		return nil, nil
+		return nil, &common.Error{Code: common.NotFound, Err: fmt.Errorf("principal not found with PrincipalFind[%+v]", find)}
```

Comment:
May I ask why we return an error when the there is no principal found?

Reply:
it's a wrong decision, I'll revert them.